ApiAnalyzer.py

```diff
@@ -0,0 +1,63 @@
+"""
+
+            ApiAnalyzer.py
+    --------------------------------
+ A class that analyze text by openai API
+ Enter your API key in the API_KEY variable
+"""
+import backoff
+import openai
+
+API_KEY = "YOUR API KEY"
+
+
+class ApiAnalyzer:
+    def __init__(self):
+        self.check_api_key()
+        self._api_key = API_KEY
+        self.chat = self.set_connection_to_api()
+
+    @staticmethod
+    def check_api_key():
+        if API_KEY == "YOUR API KEY":
+            raise Exception("Please enter your API key in the API_KEY variable in ApiAnalyzer.py")
+
+    def set_connection_to_api(self):
+        """  set the connection to the API
+        :return: the system prompt
+        """
+        openai.api_key = self._api_key
+        system_prompt = "You're an AI text analyzer assisting with presentation summarization.For each slide's content"\
+                        "you receive, generate a concise summary and additional explanation of the text, in case " \
+                        "there is phrase you are not knowing try to understand from the complete text.\n"
+        return [{"role": "system", "content": system_prompt}]
+
+    @backoff.on_exception(backoff.expo, openai.error.RateLimitError)
+    async def _get_explanation(self):
+        """ analyze the text by request to the API, return the response
+        :return: the response of the API
+        """
+        completion = openai.ChatCompletion.create(model="gpt-3.5-turbo", messages=self.chat, timeout=90)
```

Comment:
Variables like timeout should get a constant at the top of the file, or even in a separate config-file so that they can easily be configured later.